sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -79,8 +101,55 @@ export interface AccessToken {
    * The timestamp when the access token should be refreshed, in milliseconds, UNIX epoch time.
    */
   refreshAfterTimestamp?: number;
+
+  /** Type of token - Bearer or PoP */
+  tokenType?: string;
+}
+
+/**
+ * @internal
+ * @param accessToken - Access token
+ * @returns Whether a token is bearer type or not
+ */
+export function isBearerToken(accessToken: AccessToken): boolean {
+  return !accessToken.tokenType || accessToken.tokenType === "Bearer";
 }
 
+/**
+ * @internal
+ * @param accessToken - Access token
+ * @returns Whether a token is Pop token or not
+ */
+export function isPopToken(accessToken: AccessToken): boolean {
+  return accessToken.tokenType === "pop";
+}
+
+/**
+ * @internal
+ * Computes Token type "Bearer" or "Pop".
+ * Used for authentication headers.
+ * @param accessToken - Access token
+ * @returns token type
+ */
+export function computeTokenType(accessToken: AccessToken): string {
```

Comment:
nit: `getTokenType`? "compute" seems to imply some caculation.

Reply:
Yes

Reply:
done